usr/local/www/diag_logs_filter_summary.php

```diff
@@ -80,71 +84,146 @@ function stat_block($summary, $stat, $num) {
 			$numentries++;
 			$outstr = $k[$i];
 			if (is_ipaddr($outstr)) {
-				$outstr = "<a href=\"diag_dns.php?host={$outstr}\" title=\"".gettext("Reverse Resolve with DNS")."\"><img border=\"0\" src=\"/themes/{$g['theme']}/images/icons/icon_log.gif\" alt=\"log\" /></a> {$outstr}";
+				print('<tr><td>' . $outstr . '</td>' . '<td>' . $summary[$stat][$k[$i]] . '</td><td><a href="diag_dns.php?host=' . $outstr . '" class="btn btn-xs btn-success" title="' . gettext("Reverse Resolve with DNS") . '">Lookup</a></td></tr>');
+
 			} elseif (substr_count($outstr, '/') == 1) {
 				list($proto, $port) = explode('/', $outstr);
 				$service = getservbyport($port, strtolower($proto));
 				if ($service)
 					$outstr .= ": {$service}";
 			}
-			print "<tr><td>{$outstr}</td><td width=\"50\" align=\"right\">{$summary[$stat][$k[$i]]}</td></tr>";
+
+			if(!is_ipaddr($outstr))
+				print('<tr><td>' . $outstr . '</td><td>' . $summary[$stat][$k[$i]] . '</td><td></td></tr>');
 		}
 	}
 	$leftover = $gotlines - $total;
 	if ($leftover > 0) {
-		print "<tr><td>Other</td><td width=\"50\" align=\"right\">{$leftover}</td></tr>";
+		print "<tr><td>Other</td><td>{$leftover}</td><td></td>";
 	}
 	print "</table>";
+	print('</div>');
 }
 
-function pie_block($summary, $stat, $num) {
-	global $gotlines, $fields;
+// Create the JSON document for the chart to be displayed
+// Todo: Be good to investigate building this with json_encode and friends some time
+function pie_block($summary, $stat, $num, $chartnum) {
+	global $fields, $segcolors, $gotlines, $numcolors;
+
+	echo <<<PIEEND
```

Comment:
I’ll consider that. Thanks for the suggestion.

> On May 8, 2015, at 10:03 AM, SjonHortensius <notifications@github.com> wrote:
> 
> In usr/local/www/diag_logs_filter_summary.php <https://github.com/SjonHortensius/pfsense/pull/172#discussion_r29941770>:
> 
> >  }
> >  
> > -function pie_block($summary, $stat, $num) {
> > -	global $gotlines, $fields;
> > +// Create the JSON document for the chart to be displayed
> > +// Todo: Be good to investigate building this with json_encode and friends some time
> > +function pie_block($summary, $stat, $num, $chartnum) {
> > +	global $fields, $segcolors, $gotlines, $numcolors;
> > +
> > +	echo <<<PIEEND
> Why not use
> 
> ?>
> <script>
> var pie = new d3pie("pieChart <?=$chartnum?>, {
>     "header"....
> like we do for html output?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/SjonHortensius/pfsense/pull/172/files#r29941770>.
> 

Steve
--
Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com <mailto:sbeaver@netgate.com> | www.netgate.com <http://www.netgate.com/>